src/main/java/gov/cms/madie/terminology/service/VsacService.java

```diff
@@ -276,15 +277,19 @@ private void buildVsacErrorMessage(CqlCode cqlCode, VsacCode vsacCode) {
   }
 
   private Optional<CodeSystemEntry> getCodeSystemEntry(
-      List<CodeSystemEntry> codeSystemEntries, String cqlCodeSystemOid) {
+      List<CodeSystemEntry> codeSystemEntries, String cqlCodeSystemOid, String model) {
     return codeSystemEntries.stream()
-        .filter(
-            cse ->
-                cse.getUrl()
-                    .equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(cqlCodeSystemOid)))
+        .filter(cse -> isCodeSystemMatch(cse, cqlCodeSystemOid, model))
         .findFirst();
   }
 
+  private boolean isCodeSystemMatch(CodeSystemEntry cse, String oid, String model) {
+    if ("QDM".equals(model)) {
+      return cse.getOid().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
+    }
+    return cse.getUrl().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
```

Comment:
Trying to see if the `if` could be safely removed.

Perhaps a simple `or` will suffice?
```suggestion
    return cse.getOid().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid))
        || cse.getUrl().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
```

Reply:
I think that is a good approach but that way we will be allowing QDM way of declaration in FHIR and vice versa, right?

Reply:
Alas, it would. Never mind.